source/simulator/solver.cc

```diff
@@ -175,7 +176,7 @@ namespace aspect
       * This operator can either just apply the preconditioner (AMG)
       * or perform an inner CG solve with the same preconditioner.
       */
-    template <class PreconditionerA>
+    template <class PreconditionerA, class VectorType, class StokesMatrixType>
     class InverseVelocityBlock
```

Comment:
This only exists in solver.cc and won't be visible to GMG unless you move it to the header...

Reply:
maybe a first PR would only make this change and introduce the template arguments.